src/types.ts

```diff
@@ -400,3 +403,23 @@ export type Fetch = (
   input: RequestInfo | URL,
   init?: RequestInit | undefined,
 ) => Promise<Response>;
+
+export type RootOnUpdate = (
```

Comment:
Would we be better off passing the callbacks as a separate prop, and type the `onUpdate` prop with the existing `HvComponentOnUpdate` type? I think this could help preventing duplication.

Reply:
I tried using a composite but I wasn't able to get it to work. Maybe I'll take another stab.
```onUpdate = HvComponentOnUpdate & { callbacks: OnUpdateCallbacks }```
But that didn't satisfy the signature for some reason.

Reply:
I meant, the component would have two props:
```js
...
onUpdate: HvComponnetUpdate;
onUpdateCallbacks: OnUpdateCallbacks;
```

Reply:
Resolved by adding to the `HvComponentOptions`